torax/_src/imas_tools/input/core_profiles.py

```diff
@@ -0,0 +1,155 @@
+      input core/plasma_profiles IDS. Default value is True meaning that psi is
+      taken from the geometry.
+  t_initial: Initial time used to map the profiles in the dicts. If None the
+      initial time will be the time of the first time slice of the ids. Else
+      all time slices will be shifted by t_initial.
+
+  Returns:
+  Dict containing the updated fields read from the IDS that need to be replaced
+  in the input config using ToraxConfig.update_fields method.
+  """
+  profiles_1d = ids.profiles_1d
+  time_array = [float(profiles_1d[i].time) for i in range(len(profiles_1d))]
+  if t_initial:
+    time_array = [ti - time_array[0] + t_initial for ti in time_array]
+  rhon_array = [
+      profiles_1d[i].grid.rho_tor_norm for i in range(len(profiles_1d))
+  ]
+  t_initial = float(profiles_1d[0].time)
+
+  # profile_conditions
+  if not read_psi_from_geo:
+    psi = {
+        t_initial: {
+            rhon_array[0][rj]: profiles_1d[0].grid.psi[rj]
+            for rj in range(len(rhon_array[0]))
+        }
+    }
+  else:
+    psi = None
+  # Will be overwritten if Ip_from_parameters = False, when Ip is given by the equilibrium.
+  Ip = {
+      time_array[ti]: -1 * ids.global_quantities.ip[ti]
+      for ti in range(len(time_array))
+  }
+  # It is assumed the temperatures and density profiles are defined until rhon=1.
+  # Validator will raise an error if rhon[-1]!= 1.
+  T_e = {
+      time_array[ti]: {
+          rhon_array[ti][rj]: profiles_1d[ti].electrons.temperature[rj] / 1e3
+          for rj in range(len(rhon_array[ti]))
+      }
+      for ti in range(len(time_array))
+  }
+
+  if len(profiles_1d[0].t_i_average > 0):
+    T_i = {
+        time_array[ti]: {
+            rhon_array[ti][rj]: profiles_1d[ti].t_i_average[rj] / 1e3
+            for rj in range(len(rhon_array[ti]))
+        }
+        for ti in range(len(time_array))
+    }
+  else:
+    t_i_average = [
+        np.mean(
+            [
+                profiles_1d[ti].ion[iion].temperature
+                for iion in range(len(profiles_1d[ti].ion))
+            ],
+            axis=0,
+        )
+        for ti in range(len(time_array))
+    ]
+    T_i = {
+        time_array[ti]: {
+            rhon_array[ti][rj]: t_i_average[rj] / 1e3
+            for rj in range(len(rhon_array[ti]))
+        }
+        for ti in range(len(time_array))
+    }
+
+  n_e = {
+      time_array[ti]: {
+          rhon_array[ti][ri]: profiles_1d[ti].electrons.density[ri]
+          for ri in range(len(rhon_array[ti]))
+      }
+      for ti in range(len(time_array))
+  }
+
+  if len(
+      ids.global_quantities.v_loop > 0
+  ):  # Map v_loop_lcfs in case it is used as bc for psi equation.
+    v_loop_lcfs = {
+        time_array[ti]: ids.global_quantities.v_loop[ti]
+        for ti in range(len(time_array))
+    }  # TODO: Check the sign for v_loop when it will be used.
+  else:
+    v_loop_lcfs = [0.0]
+
+  return {
+      'profile_conditions.Ip': Ip,
```

Comment:
Do we want to return it like this or more of a nested dict to then use to update the config before it is created as was discussed in the doc